---
title: "Case study 2: Allergic Asthma"
output:
  rmdformats::readthedown:
    self_contained: true
    thumbnails: true
    lightbox: true
    gallery: false
    highlight: tango
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, tidy = TRUE)

WhereAmI <- "~/Dropbox/PROOF/Manuscript/mixOmics/diablo/analyses/casestudy2_asthma/"

# load libraries
library(corrplot)
library(pROC)
library(amritr)
library(tidyverse)
library(mixOmics)
library(NMF)
library(RColorBrewer)
library(cowplot)
library(scales)
library(knitr)
library(UpSetR)
library(org.Hs.eg.db)
library(KEGG.db)
library(annotate)
library("hugene10sttranscriptcluster.db")
library(grid)
source(paste0(WhereAmI, "asthma_functions.R"))

## load data
load("~/Dropbox/PROOF/Manuscript/mixOmics/diablo_datasets/asthma/asthmaDatasets.RDATA")
## clean datasets
cells <- cells[, c("Relative.Neutrophils","Relative.Lymphocytes","Relative.Monocytes",
  "Relative.Eosinophils","Relative.Basophils","Treg","Tcells","Bcells","Th17")]
metExp <- metExp[apply(metExp, 1, mad) >0, ]
rownames(metExp) <- metabolites[rownames(metExp), "BIOCHEMICAL"]

## identify any overlapping gene and metabolite modules names
colnames(gene.module)[colnames(gene.module) %in% intersect(colnames(gene.module), colnames(metabolite.module))] <- 
  paste("genMod", colnames(gene.module)[colnames(gene.module) %in% intersect(colnames(gene.module), colnames(metabolite.module))], sep = "-")

```

## DIABLO for repeated measures designs and module-based analyses 

Next, we demonstrate the flexibility of DIABLO by extending its use to a repeated measures cross-over study [27], as well as incorporating module-based analyses that incorporate prior biological knowledge [28–30]. We use a small multi-omics asthma dataset, including pre and post intervention timepoints, to compare a DIABLO model that can account for repeated measures (multilevel DIABLO) with the standard DIABLO model as described above [20,21]. An allergen inhalation challenge was performed as we previously described in [20,21] in 14 subjects and blood samples were collected before (pre) and two hours after (post) challenge; cell-type frequencies, leukocyte gene transcript expression and plasma metabolite abundances were determined for all samples. 

```{r out.width="100%"}
img1_path <- "/Users/asingh/Dropbox/PROOF/Manuscript/mixOmics/diablo/analyses/casestudy1_brca/Figures/Table1.png"
include_graphics(img1_path)
```

*Overview of multi-omics datasets analyzed for method benchmarking and in two case studies. The breast cancer case study includes training and test datasets for all omics types except proteins.*

## FEV1 profiles

We observed a net decline in lung function after allergen inhalation challenge (Supplementary Fig. 9), and the goal of this study was to identify perturbed molecular mechanisms in the blood in response to allergen inhalation challenge. 

```{r fev1Plots, fig.path='Figures/', dev='png', fig.height = 5, fig.width = 7}
fev1 <- read.csv("~/Dropbox/PROOF/Manuscript/mixOmics/diablo_datasets/asthma/fev1Data.csv", row.names = 1)[, c("BLFEV","F10L","F20L","F30L","F45L","F60L","F90L","F120L","F180L","F240L","F300L","F360L","F420L")]

p <- scale(t(fev1), center = fev1$BLFEV, scale = fev1$BLFEV) %>% as.data.frame() %>%
  tbl_df() %>%
  mutate(Time = c(0, 10, 20, 30, 45, 60, 90, 120, 180, 240, 300, 360, 420)) %>%
  tidyr::gather(Subject, FEV1, -Time) %>%
  filter(Time %in% c(0, 10, 20, 30, 45, 60, 90, 120)) %>%
  mutate(Time = Time/60) %>% 
  mutate(fev1 = 100*FEV1) %>%
  ggplot(aes(x = Time, y = fev1, group = Subject)) +
  geom_point() + geom_line(color="black") +
  scale_y_continuous(expression('Percent drop in '~ FEV[1])) + theme_bw() +
  scale_x_continuous(expression('Time (hours)')) +
  theme(axis.text.y = element_text(size = 15, hjust = 1)) + theme(axis.text.x = element_text(size = 15, hjust = 0.5))+
  theme(axis.title.x=element_text(size = 15)) + theme(axis.title.y=element_text(size = 15,angle = 90))+ 
  theme(legend.key = element_rect(colour = "black", fill="white"))  +
  theme(plot.background = element_rect()) +  
  theme(strip.background = element_rect(colour = "black", fill = "white",
                                        size = 1), strip.text.x = element_text(size=20)) +
  geom_hline(yintercept = 0, colour="yellow3", linetype = "longdash") +
  theme(legend.position="none")

p
```

## DIABLO

A module based approach (also known as eigengene summarization [18], see Methods) was used to transform both the gene expression and metabolite datasets into pathway datasets. Consequently, each variable in those two datasets now represented the scaled pathway activity expression level for each sample instead of direct gene/metabolite expression. The mRNA dataset was transformed into a dataset of metabolic pathways (based on the Kyoto Encyclopedia of Genes and Genomes, KEGG) whereas the metabolite dataset was transformed into a metabolite pathway dataset based on annotations provided by Metabolon Inc. (Durham, North Carolina, USA). 

```{r out.width="100%"}
img1_path <- "/Users/asingh/Dropbox/PROOF/Manuscript/mixOmics/diablo/analyses/casestudy2_asthma/Figures/designMatrix.png"
include_graphics(img1_path)
```

To account for the repeated measures experimental design, a multilevel approach [27] was first used to isolate the within-sample variation from each dataset (see Methods), and then DIABLO was applied to identify a multi-omics biomarker panel consisting of cells, gene and metabolite modules that discriminated pre- from post-challenge samples. 

## tune keepX

```{r}
X <- list(cells=cells, gene.module=gene.module, metabolite.module=metabolite.module)
time <- demo$Time
Cov <- data.frame(sample = rep(1:14, 2), time = demo$Time)
A = lapply(X, function(i) suppressMessages(withinVariation(X = i, design = Cov)))

lapply(A, dim)

## run DIABLO
ncomp <- 2
design <- matrix(1, nrow = 3, ncol = 3)
diag(design) <- 0
```

## error rate of optimal keepX

We contrast the resulting ‘multilevel DIABLO’ (mDIABLO) with a standard DIABLO model that disregards the paired nature of this study by comparing their cross-validation classification performances.

```{r aucPlot, fig.path='Figures/', dev='png', fig.height = 7, fig.width = 7}
keepX = list(cells = c(3, 3), gene.module = c(5, 5), metabolite.module = c(5, 5))
ncomp = 2
sample = rep(1:14, 2)
time = demo$Time
Y = time

cv <- leave.one.out.cv(X, Y, ncomp, keepX, design, sample)

aucPlot <- rbind(cv$unpaired$roc.res, cv$paired$roc.res) %>% 
  mutate(Time = c(rep("DIABLO", nrow(cv$unpaired$roc.res)), rep("mDIABLO", nrow(cv$paired$roc.res)))) %>% 
  ggplot(aes(x = fp, y = tp, fill = Time, color = Time)) +
  geom_abline(intercept = 0, slope = 1, col = "gray", linetype="dashed")  +
  geom_point() +
  geom_line() +
  customTheme(sizeStripFont = 15, xAngle = 0, hjust = 0.5, 
    vjust = 0.5, xSize = 20, ySize = 20, xAxisSize = 20, yAxisSize = 20) +
  xlab("100 - Specificity") + ylab("Sensitivity") +
  annotate("text", label = cv$unpaired$auc, x = 75, y = 50, size = 3.5, colour = hue_pal()(2)[1]) +
  annotate("text", label = cv$paired$auc, x = 35, y = 85, size = 3.5, colour = hue_pal()(2)[2]) +
  theme(legend.position = c(0.7, 0.15))
aucPlot
```

## DIABLO - unpaired (DIABLO) vs paired (mDIABLO)

```{r}
# unpaired
diablo = block.splsda(X = X, Y = time, ncomp = ncomp, keepX = keepX, design = design)
diabloPanel <- list(cells = c(selectVar(diablo, comp = 1)$cells$name, 
                              selectVar(diablo, comp = 2)$cells$name),
                           gene.module = c(selectVar(diablo, comp = 1)$gene.module$name,  
                                           selectVar(diablo, comp = 2)$gene.module$name),
                           metabolite.module = c(selectVar(diablo, comp = 1)$metabolite.module$name, 
                                                 selectVar(diablo, comp = 2)$metabolite.module$name))

# paired
diablow = block.splsda(X = A, Y = time, ncomp = ncomp, keepX = keepX, design = design)
diablowPanel <- list(cells = c(selectVar(diablow, comp = 1)$cells$name, 
                              selectVar(diablow, comp = 2)$cells$name),
                           gene.module = c(selectVar(diablow, comp = 1)$gene.module$name,  
                                           selectVar(diablow, comp = 2)$gene.module$name),
                           metabolite.module = c(selectVar(diablow, comp = 1)$metabolite.module$name, 
                                                 selectVar(diablow, comp = 2)$metabolite.module$name))

```

## Component plots

mDIABLO outperformed DIABLO (AUC=98.5% vs. AUC=62.2%, leave-one-out cross-validation, see Methods), and we observed a greater degree of separation between the pre- and post-challenge samples for mDIABLO compared to DIABLO. 

```{r auc_componentPlot, fig.path='Figures/', dev='png', fig.height = 4, fig.width = 13}
dat <- as.data.frame(Reduce("+", diablo$variates)/length(X)) %>% 
  mutate(time = time, sample = factor(sample))
p1 <- ggplot(dat, aes(x = `comp 1`, y = `comp 2`, group = time, color = time)) +
  geom_point(size = 4) +
  stat_ellipse() +
  geom_line(aes(group = sample), color = "gray")+
  customTheme(sizeStripFont = 15, xAngle = 0, hjust = 0.5, vjust = 0.5, 
              xSize = 20, ySize = 20, xAxisSize = 20, yAxisSize = 20) +
  xlab("Consensus Component 1") +
  ylab("Consensus Component 2") +
  ggtitle("DIABLO") +
  theme(legend.position = c(0.1, 0.9)) + 
  scale_color_manual(values=c("#388ECC", "#F68B33"))
  

dat <- as.data.frame(Reduce("+", diablow$variates)/length(X)) %>% 
  mutate(time = time, sample = factor(sample))
p2 <- ggplot(dat, aes(x = `comp 1`, y = `comp 2`, group = time, color = time)) +
  geom_point(size = 4) +
  stat_ellipse() +
  geom_line(aes(group = sample), color = "gray")+
  customTheme(sizeStripFont = 15, xAngle = 0, hjust = 0.5, vjust = 0.5, 
              xSize = 20, ySize = 20, xAxisSize = 20, yAxisSize = 20) +
  xlab("Consensus Component 1") +
  ylab("Consensus Component 2") +
  ggtitle("mDIABLO") +
  theme(legend.position = "none") + 
  scale_color_manual(values=c("#388ECC", "#F68B33"))

plot_grid(aucPlot, p1, p2, ncol = 3)

```

## Compare DIABLO vs. mDIABLO multi-omic panels

Common features (pathways) were identified across omics-types in the mDIABLO model, but not in the standard DIABLO model. Tryptophan metabolism and Valine, leucine and isoleucine metabolism pathways were identified in both the gene and metabolite module datasets using mDIABLO. 

```{r overlap, fig.path='Figures/', dev='png', fig.height = 4, fig.width = 7}
diablowPanel_modif <- diablowPanel
diablowPanel_modif$gene.module[diablowPanel_modif$gene.module == "Valine, leucine and isoleucine biosynthesis"] <- "Valine, leucine and isoleucine metabolism"
diablowPanel_modif$gene.module[diablowPanel_modif$gene.module == "genMod-Tryptophan metabolism"] <- "Tryptophan metabolism"

panels <-  list(DIABLO_Cells = diabloPanel$cells, 
                   `DIABLO_Gene modules` = diabloPanel$gene.module, 
                   `DIABLO_Metabolite modules` = diabloPanel$metabolite.module,
                   mDIABLO_Cells = diablowPanel_modif$cells, 
                   `mDIABLO_Gene modules` = diablowPanel_modif$gene.module, 
                   `mDIABLO_Metabolite modules` = diablowPanel_modif$metabolite.module)

intersect(panels$`mDIABLO_Gene modules`, panels$`mDIABLO_Metabolite modules`)

input <- fromList(panels)
metadata <- data.frame(panels=names(panels),
  type = c(rep("DIABLO", 3), rep("mDIABLO", 3)))

upset(input, sets = colnames(input), keep.order = TRUE,
    queries = list(list(query = intersects, params = list("mDIABLO_Gene modules","mDIABLO_Metabolite modules"), active = TRUE, color = "maroon"),
      list(query = intersects, params = list("mDIABLO_Gene modules","mDIABLO_Metabolite modules","DIABLO_Metabolite modules"), active = TRUE, color = "maroon")),
  set.metadata = list(data = metadata, plots = list(list(type = "matrix_rows", 
    column = "type", colors = c(DIABLO = "#FBB4AE", DIABLOw = "#B3CDE3"), 
    alpha = 0.5))))

```

## Heatmap

The heatmap of pairwise associations of all features identified with mDIABLO demonstrated the ability of DIABLO to select groups of correlated features which were predictive of pre- and post-challenge samples. 

```{r varCor, fig.path='Figures/', dev='png', fig.height = 15, fig.width = 15}
corMat <- mapply(function(x, y){
  x[, unique(y)]
}, x = A, y = diablowPanel) %>% 
  do.call(cbind, .) %>% 
  cor

color.blocks = brewer.pal(n = 12, name = "Set2")[1:3]
row.sideColors = rep(color.blocks, sapply(lapply(diablowPanel, unique), length))
col.sideColors = row.sideColors

cim(corMat, margins = c(35, 35), row.cex = 2, col.cex = 2, 
  row.sideColors=row.sideColors, col.sideColors=col.sideColors)
legend("topright", names(diablowPanel), col = color.blocks, pch = 19, bty = "n", cex = 2)

```

The Asthma pathway was also identified [even though individual gene members were not significantly altered post-challenge and was negatively associated with Butanoate metabolism and positively associated with basophils, a hallmark cell-type in asthma. These findings depict DIABLO’s flexibility and sensitivity to detect subtle differences between repeated designs, and its ability to identify common molecular processes spanning different biological layers. The biological pathways identified suggest a mechanistic link with response to allergen challenge. 

## Circosplot

```{r circosPlot, fig.path='Figures/', dev='png', fig.height = 15, fig.width = 15}
circosPlot(diablow, cutoff = 0.8, showIntraLinks = FALSE, size.variables = 1, size.labels = 2, size.legend = 2, color.blocks = color.blocks)
```

## Asthma KEGG pathway

```{r Asthma_pathwayGenes, fig.path='Figures/', dev='png', fig.height = 7, fig.width = 7}
## obtain KEGG id for the Asthma pathway
xx <- as.list(KEGGPATHID2NAME)
hsaID <- unlist(xx)[unlist(xx)=="Asthma"]

## extract the entrez gene ids for members of the Asthma pathway
kegg <- org.Hs.egPATH2EG
entrez <- unlist(as.list(kegg[names(hsaID)]))

## map entrez ids to affymetrix probe set ids
eg <- getEG(rownames(genExp), "hugene10sttranscriptcluster")
geneIDs <- eg[eg %in% entrez]

## extract expression data for genes in the asthma pathway
genesDat <- genExp[names(geneIDs), ]

pval <- apply(genesDat, 1, function(i){
  t.test(i[demo$Time == "post"], i[demo$Time == "pre"], paired = TRUE)$p.value
})
padj <- p.adjust(pval, "BH")
fc <- apply(genesDat, 1, function(i){
 mean(i[demo$Time == "post"]) - mean(i[demo$Time == "pre"])
})


plot(-log10(pval) ~ fc, pch = 19, col = 2, xlim = c(-0.17, 0.155),
     ylab=expression("-log"[10]~~~"p-value"), xlab = expression("log"[2]~~~"fold-change"),
     main = "Asthma KEGG pathway")
points(-log10(pval) ~ fc, pch = 21)
text(x = fc, y = -log10(pval), labels = getSYMBOL(names(pval), "hugene10sttranscriptcluster"),
     cex = 0.6, pos = 1:4)
abline(h = -log10(0.05), lty = 2)
text(x = -0.10, y = -log10(0.045), labels = "P-value = 0.05")

```

## References

3. Wang B, Mezlini AM, Demir F, Fiume M, Tu Z, Brudno M, et al. Similarity network fusion for aggregating data types on a genomic scale. Nat Methods [Internet]. 2014 [cited 2016 Jan 19];11:333–7. Available from: http://www.nature.com/doifinder/10.1038/nmeth.2810
18. Langfelder P, Horvath S. WGCNA: an R package for weighted correlation network analysis. BMC Bioinformatics [Internet]. 2008 [cited 2016 Apr 4];9:559. Available from: http://www.biomedcentral.com/1471-2105/9/559
19. The TCGA Research Network. The Cancer Genome Atlas [Internet]. Available from: http://cancergenome.nih.gov/
20. Singh A, Yamamoto M, Kam SHY, Ruan J, Gauvreau GM, O’Byrne PM, et al. Gene-metabolite expression in blood can discriminate allergen-induced isolated early from dual asthmatic responses. Hsu Y-H, editor. PLoS ONE [Internet]. 2013 [cited 2015 Jul 18];8:e67907. Available from: http://dx.plos.org/10.1371/journal.pone.0067907
21. Singh A, Yamamoto M, Ruan J, Choi JY, Gauvreau GM, Olek S, et al. Th17/Treg ratio derived using DNA methylation analysis is associated with the late phase asthmatic response. Allergy Asthma Clin Immunol [Internet]. 2014 [cited 2016 Mar 2];10:32. Available from: http://www.biomedcentral.com/content/pdf/1710-1492-10-32.pdf
27. Liquet B, Lê Cao K-A, Hocini H, Thiébaut R. A novel approach for biomarker selection and the integration of repeated measures experiments from two assays. BMC Bioinformatics [Internet]. 2012 [cited 2015 Jul 18];13:325. Available from: http://www.biomedcentral.com/1471-2105/13/325/
28. Allahyar A, de Ridder J. FERAL: network-based classifier with application to breast cancer outcome prediction. Bioinformatics [Internet]. 2015 [cited 2018 Feb 1];31:i311–9. Available from: https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btv255
29. Cun Y, Fröhlich H. Network and data integration for biomarker signature discovery via network smoothed t-statistics. Boccaletti S, editor. PLoS ONE [Internet]. 2013 [cited 2017 May 30];8:e73074. Available from: http://dx.plos.org/10.1371/journal.pone.0073074
30. Sokolov A, Carlin DE, Paull EO, Baertsch R, Stuart JM. Pathway-based genomics prediction using generalized elastic net. PLoS Comput Biol [Internet]. 2016 [cited 2017 May 30];12:e1004790. Available from: http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004790
